src/coreclr/nativeaot/Runtime/thread.h

```diff
@@ -99,6 +79,7 @@ struct ThreadBuffer
 #endif // FEATURE_HIJACK
     PTR_ExInfo              m_pExInfoStackHead;
     Object*                 m_threadAbortException;                 // ThreadAbortException instance -set only during thread abort
+    gc_alloc_context*       m_palloc_context;                       // pointer to the allocation context for the current thread.
```

Comment:
This indirection is unnecessary for native AOT. I would get rid of it and keep alloc context as the first thing in the ThreadBuffer like before.

The indirection is necessary for regular CoreCLR for now since the `Thread` is everything in kitchen sink that outlives the physical thread. I would hope that we can eventually make Thread in regular CoreCLR to be TLS static like it is in native AOT.

Reply:
Sounds good. I'll revert the NativeAOT changes other than retyping the buffer as a `gc_alloc_context` instead of a byte array of the same size.

Reply:
Actually, I'll keep it as-is in main since we can't easily include gcinterface.h in this location.